packages/uilib/src/lib/plugins/type-designer/editor-events/editor-event-handler.ts

```diff
@@ -0,0 +1,87 @@
+import { CreateBayEvent, CreateIEDEvent, CreateLDeviceEvent, CreateSubstationEvent, CreateVoltageLevelEvent } from "./event-types"
+import type { Create, Delete } from "./editor-events"
+import type { BayType, IEDType, LDeviceType, VoltageLevelType, SubstationType } from "../types"
+import { getNodeName } from "../components";
+
+export class EditorEventHandler {
+    private readonly editorActionName = "editor-action"
+
+    constructor(private readonly root: HTMLElement, private readonly dataTemplates: Element) {
+    }
+
+    public dispatchCreateBay(event: CreateBayEvent): void {
+        this.dispatchCreate(event);
+    }
+
+    public dispatchCreateSubstation(event: CreateSubstationEvent): void {
+        this.dispatchCreate(event);
+    }
+
+    public dispatchCreateLDevice(event: CreateLDeviceEvent): void {
+        this.dispatchCreate(event);
+    }
+
+    public dispatchCreateIED(event: CreateLDeviceEvent): void {
+        this.dispatchCreate(event);
+    }
+
+    public dispatchCreateVoltageLevel(event: CreateLDeviceEvent): void {
+        this.dispatchCreate(event);
+    }
+
+    private dispatchCreate(event: CreateBayEvent | CreateIEDEvent | CreateVoltageLevelEvent | CreateLDeviceEvent | CreateSubstationEvent): void {
+        const newNode = this.buildNewNode(event.type);
+        const replaces = this.buildCreateEvents(newNode);
+        const combinedEditorEvent = this.buildEditorActionEvent(replaces);
```

Comment:
There is no reason use a combined event here, we only use a single `Create`, see the Open SCD Edit API on how to build a single event

Reply:
Use of OpenSCD API.